src/language-js/utils.js

```diff
@@ -1473,6 +1422,70 @@ function hasIgnoreComment(path) {
   return hasNodeIgnoreComment(node);
 }
 
+const CommentCheckFlags = {
+  leading: 1 << 1,
+  trailing: 1 << 2,
+  dangling: 1 << 3,
+  block: 1 << 4,
+  line: 1 << 5,
+  prettierIgnore: 1 << 6,
+  first: 1 << 7,
+  last: 1 << 8,
+};
+const getCommentTestFunction = (types, fn) => {
+  if (typeof types === "function") {
+    fn = types;
+    types = 0;
+  }
+  if (types || fn) {
+    return (comment, index, comments) =>
+      !(
+        (types & CommentCheckFlags.leading && !comment.leading) ||
+        (types & CommentCheckFlags.trailing && !comment.trailing) ||
+        (types & CommentCheckFlags.dangling &&
+          (comment.leading || comment.trailing)) ||
+        (types & CommentCheckFlags.block && !isBlockComment(comment)) ||
+        (types & CommentCheckFlags.line && !isLineComment(comment)) ||
+        (types & CommentCheckFlags.first && index !== 0) ||
+        (types & CommentCheckFlags.last && index !== comments.length - 1) ||
+        (types & CommentCheckFlags.prettierIgnore &&
+          !isPrettierIgnoreComment(comment)) ||
+        (fn && !fn(comment))
+      );
+  }
+};
+/**
+ * @param {Node} node
+ * @returns {boolean}
+ */
+function hasComment(node, types, fn) {
```

Comment:
?
```suggestion
function hasComment(node, flags, fn) {
```

Reply:
Also should be this argument named `flag` rather than `flags`, as it takes a single flag?

Reply:
It can take multiple. That's the point of bitmasks: `CommentCheckFlags.Leading | CommentCheckFlags.Block`.

Reply:
I got it!

Reply:
`{ leading: true, block: true }` would be a friendlier API, but presumably it's worse performance-wise.